perllib/FixMyStreet/Cobrand/Bexley/Garden.pm

```diff
@@ -50,26 +52,61 @@ sub garden_current_subscription {
     my $contract = $customer->{ServiceContracts}[0];
     return unless $contract;
 
-    my $parser
-        = DateTime::Format::Strptime->new( pattern => '%d/%m/%Y %H:%M' );
-    my $end_date = $parser->parse_datetime( $contract->{EndDate} );
+    # XXX should maybe sort by CreatedDate rather than assuming first is OK
+    $sub->{cost} = try {
+        my ($payment) = grep { $_->{PaymentStatus} eq 'Paid' } @{ $contract->{Payments} };
+        return $payment->{Amount};
+    };
+
+    my $parser = DateTime::Format::Strptime->new( pattern => '%d/%m/%Y %H:%M' );
+    $sub->{end_date} = $parser->parse_datetime( $contract->{EndDate} );
+
+    $sub->{customer_external_ref} = $customer->{CustomerExternalReference};
+
+    $sub->{bins_count} = $contract->{WasteContainerQuantity};
+
+    my $c = $self->{c};
+    my $p = $c->model('DB::Problem')->search({
+        category => 'Garden Subscription',
+        extra => { '@>' => encode_json({ "_fields" => [ { name => "uprn", value => $uprn } ] }) },
+        state => { '!=' => 'hidden' },
+        external_id => "Agile-" . $contract->{Reference},
+    })->order_by('-id')->to_body($c->cobrand->body)->first;
+
+    $sub->{row} = $p if $p;
+
+    return $sub;
+}
+
+sub garden_current_subscription {
+    my ($self, $services) = @_;
+
+    my $current = $self->{c}->stash->{property}{garden_current_subscription};
+    return $current if $current;
+
+    my $uprn = $self->{c}->stash->{property}{uprn};
+    return undef unless $uprn;
 
     # Agile says there is a subscription; now get service data from
     # Whitespace
-    my $services = $self->{c}->stash->{services};
+    my $service_ids = { map { $_->{service_id} => $_ } @$services };
     for ( @{ $self->garden_service_ids } ) {
-        if ( my $srv = $services->{$_} ) {
-            $srv->{customer_external_ref}
-                = $customer->{CustomerExternalReference};
-            $srv->{end_date} = $end_date;
+        if ( my $srv = $service_ids->{$_} ) {
+            my $sub = $self->lookup_subscription_for_uprn($uprn);
+            $srv->{customer_external_ref} = $sub->{customer_external_ref};
+            $srv->{end_date} = $sub->{end_date};
+            $srv->{garden_bins} = $sub->{bins_count};
+            $srv->{garden_cost} = $sub->{cost};
+            $self->{c}->stash->{orig_sub} = $sub->{row};
```

Comment:
Should this be set in `lookup_subscription_for_uprn`, at the `$sub->{row} = $p if $p;` bit?